File: 프로그래머스/2/42578. 의상/의상.cpp
#include <bits/stdc++.h>
using namespace std;

int solution(vector<vector<string>> clothes) {
    ios::sync_with_stdio(0);
    cin.tie(0);
    
    map<string, int> ma;
    for(auto c : clothes) {
        ma[c[1]]++;
    }
    
    int answer = 1;
    for(auto c : ma) {
        answer *= (c.second + 1);
    }
    return answer - 1;
}

File: 프로그래머스/2/42578. 의상/의상.java
import java.util.*;

class Solution {
    public int solution(String[][] clothes) {        
        Map<String, Integer> map = new HashMap<>();
        for (var c : clothes) {
            map.put(c[1], map.getOrDefault(c[1], 0) + 1);
        }
        
        int answer = 1;
        for (var m : map.values()) {
            answer *= (m + 1);
        }
        
        return answer - 1;
    }
}